File: libs/dablja-worker/dablja_worker/metrics.py
"""Prometheus metrics shared by pipeline workers."""
from __future__ import annotations

from prometheus_client import Counter, Histogram

STAGE_DURATION = Histogram(
    "dablja_stage_duration_seconds",
    "Wall-clock time to process a pipeline stage job",
    ["stage"],
    buckets=(1, 5, 10, 30, 60, 120, 300, 600, 1200, 1800, 3600, 7200),
)

JOBS_COMPLETED = Counter(
    "dablja_jobs_completed_total",
    "Pipeline jobs completed by stage and terminal status",
    ["stage", "status"],
)

DLQ_MESSAGES = Counter(
    "dablja_dlq_messages_total",
    "Poison messages observed on the orchestrator DLQ",
)

File: orchestrator/internal/metrics/metrics.go
package metrics

import (
	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promauto"
)

var (
	JobsCompleted = promauto.NewCounterVec(prometheus.CounterOpts{
		Name: "dablja_jobs_completed_total",
		Help: "Pipeline jobs completed by orchestrator stage transition and status",
	}, []string{"stage", "status"})

	StageDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
		Name:    "dablja_stage_duration_seconds",
		Help:    "Orchestrator message handler duration in seconds",
		Buckets: []float64{0.01, 0.05, 0.1, 0.5, 1, 2, 5, 10, 30, 60},
	}, []string{"handler"})

	DLQMessages = promauto.NewCounter(prometheus.CounterOpts{
		Name: "dablja_dlq_messages_total",
		Help: "Messages nacked without requeue and routed to the DLQ",
	})
)
